core/src/main/java/org/elasticsearch/common/util/BigArrays.java

```diff
@@ -435,15 +440,18 @@ public CircuitBreakerService breakerService() {
 
     private <T extends AbstractBigArray> T resizeInPlace(T array, long newSize) {
         final long oldMemSize = array.ramBytesUsed();
+        assert oldMemSize == array.ramBytesEstimated(array.size) :
+            "ram bytes used should equal that which was previously estimated";
+        final long estimatedIncreaseInBytes = array.ramBytesEstimated(newSize) - oldMemSize;
```

Comment:
Can we add an assertion that this is never negative? I don't think it ever will be, but just to be sure...